controllers/cluster.weave.works/existinginframachine_controller.go

```diff
@@ -483,6 +486,27 @@ func (a *ExistingInfraMachineReconciler) update(ctx context.Context, c *existing
 	if err = a.setNodeProviderIDIfNecessary(ctx, node); err != nil {
 		return err
 	}
+	isMaster := isMaster(node)
+	if isMaster {
+		// Check if the kubeadm-certs secret exists.
+		// If not, run the renewal plan to upload new certs:
+		// kubeadm init phase upload-certs --upload-certs
+		// and update wks-controller-secrets with the new certificate key and bootstrap token ID
+		// so that new control plane nodes can be added to the cluster.
+
+		exists, _ := a.kubeadmCertsExists(ctx)
```

Comment:
Can the certs exist and have expired?

Reply:
The `kubeadm-certs` have an owner reference to a bootstrap-token, so they are removed once the token expires, https://pwittrock.github.io/docs/admin/bootstrap-tokens/#token-management-with-kubeadm

> These Secrets are then read by the Bootstrap Authenticator in the API Server. Expired tokens are removed with the TokenCleaner controller in the Controller Manager.

There might be a small time gap from when they expire until they are removed but that should cause the node join to fail, and in a following loop the wks-controller will recreate it once it reconciles a control plane node.